std/js/Lib.hx

```diff
@@ -122,4 +122,15 @@ class Lib {
 	@:pure(false) public static function rethrow() {
 		// function is implemented in the compiler
 	}
+
+	/**
+		Get original caught exception object, before unwrapping the `js.Boot.HaxeError`.
+
+		Can be useful if we want to redirect the original error into some external API (e.g. Promise or node.js callbacks).
+
+		Calling this is only possible inside a catch statement.
+	**/
+	public static function getOriginalException():Dynamic {
```

Comment:
This one allows for implementing functions like this:

```haxe
static function catchToCallback<T>(fn:()->T, callback:(error:Any, result:Null<T>)->Void) {
    var result =
        try
            fn()
        catch (e:Any)
            return callback(js.Lib.getOriginalException(), null);
    callback(null, result);
}
```
which will be compiled nicely into
```js
Main.catchToCallback = function(fn,callback) {
    var result;
    try {
        result = fn();
    } catch( e ) {
        callback(e,null);
        return;
    }
    callback(null,result);
};
```
(note how we don't unwrap `e` here, because we requested "original exception")

Reply:
Not really, it's just not generated in a simple case where `haxe.CallStack` is not used.